tests/Reflection/ClassReflectionTest.php

```diff
@@ -25,30 +26,108 @@ public function testGetDeclaredMethods(): void
 		Assert::assertSame('barMethod', $methods[0]->name);
 	}
 
-	public function testHasDeclaredMethod(): void
+	/**
+	 * @return mixed[][]|\Generator
+	 */
+	public function hasDeclaredMethodWithSensitivityDataProvider(): Generator
 	{
-		$classReflection = new ReflectionClass(Bar::class);
-		Assert::assertTrue(ClassReflection::hasDeclaredMethod($classReflection, 'barMethod'));
-		Assert::assertFalse(ClassReflection::hasDeclaredMethod($classReflection, 'fooMethod'));
-		Assert::assertFalse(ClassReflection::hasDeclaredMethod($classReflection, 'xxx'));
-		Assert::assertTrue(ClassReflection::hasDeclaredMethod($classReflection, 'barMethod', ClassReflection::CASE_SENSITIVE));
-		Assert::assertFalse(ClassReflection::hasDeclaredMethod($classReflection, 'barmethod', ClassReflection::CASE_SENSITIVE));
+		yield 'method exists' => [
+			'classFqn' => Bar::class,
+			'methodName' => 'barMethod',
+			'expectedCaseSensitiveResult' => true,
+			'expectedCaseInsensitiveResult' => true,
+		];
+		yield 'method exists, but name is in lowercase' => [
+			'classFqn' => Bar::class,
+			'methodName' => 'barmethod',
+			'expectedCaseSensitiveResult' => false,
+			'expectedCaseInsensitiveResult' => true,
+		];
+		yield 'method does not exist' => [
+			'classFqn' => Bar::class,
+			'methodName' => 'fooMethod',
+			'expectedCaseSensitiveResult' => false,
+			'expectedCaseInsensitiveResult' => false,
+		];
+		yield 'odd method name' => [
```

Comment:
The `xxx/XXX` values actually cover the case of `nonexistent method/property/constant` proper, whereas the `foo/FOO` values cover the test case of `private method/property in parent class`, or `public constant in parent class`, respectively. I only found this out because of failing coverage after removing them. So, instead of removing the test cases as duplicate, I rather named them correctly, if you agree?